elixir/apps/domain/lib/domain/gateways.ex

```diff
@@ -147,6 +158,16 @@ defmodule Domain.Gateways do
     {:ok, gateways}
   end
 
+  def list_gateways_for_resource(%Resources.Resource{} = resource) do
+    gateways =
+      Gateway.Query.all()
+      |> Gateway.Query.by_account_id(resource.account_id)
+      |> Gateway.Query.by_resource_id(resource.id)
+      |> Repo.all()
+
+    {:ok, gateways}
```

Comment:
We have `Repo.list/1` just so that we don't need to add `{:ok, tuple}` in all the functions.
```suggestion
    Gateway.Query.all()
    |> Gateway.Query.by_account_id(resource.account_id)
    |> Gateway.Query.by_resource_id(resource.id)
    |> Repo.list()
```

Reply:
👍 Makes sense